ex2/test2.sql

```diff
@@ -0,0 +1,121 @@
+	user_id bigint not null auto_increment,
+    username varchar(16) not null unique,
+    email varchar(50) not null unique,
+    password varchar(10) not null,
+    birthday date,
+    image_url text,
+    is_active boolean not null,
+    primary key(user_id),
+    created_at timestamp not null default current_timestamp,
+    update_at timestamp not null default current_timestamp
+);
+create table test2.project_users(
+	project_user_id bigint not null auto_increment,
+    project_id bigint not null,
+    user_id bigint not null,
+    created_at timestamp not null default current_timestamp,
+    update_at timestamp not null default current_timestamp,
+    primary key(project_user_id),
+    constraint fk_projects_users_projects foreign key(project_id) references projects(project_id),
+    constraint fk_projects_users_users foreign key(user_id) references users(user_id)
+);
+-- Câu 2. Viết lệnh sql để tạo dữ liệu cho các bảng
+INSERT INTO test2.categories(category_name) VALUES('Danh muc A'),('Danh muc B'),('Danh muc C'),('Danh muc D');
+INSERT INTO test2.companies(company_name, company_code) values('Cong Ty A','CT001'),('Cong Ty B','CT002'),('Cong Ty C','CT003');
+INSERT INTO test2.projects(category_id, company_id,project_name,projected_spend,projected_variance,revenue_recognised) values
+(5,3,'tham mi',70,30,1000),
+(5,1,'tai chinh',50,20,600),
+(5,2,'giao duc',60,20,1000),
+(6,1,'kinh doanh',60,10,1000),
+(6,2,'thoi trang',60,40,800),
+(7,3,'tham mi',70,30,1000);
+
+INSERT INTO test2.users(username,email,password,birthday,image_url,is_active) values
+('Tran Le Manh','abc@gmail.com','1234','2001-10-10','temp.jpg',1),
+('Tran Van Dung','def@gmail.com','1234','2001-10-20','temp2.jpg',0),
+('Tran Van Nam','ghk@gmail.com','1212','2001-10-16','temp4.jpg',1),
+('Tran Van Trung','abcd@gmail.com','12234','2001-08-10','temp8.jpg',1),
+('Tran Van Thanh','itjg@gmail.com','23434','2001-06-10','temp23.jpg',1),
+('Tran Le Duy','ggfs@gmail.com','4356','2001-02-10','temp45.jpg',0);
+INSERT INTO test2.project_users(project_id,user_id) values
+(12,4),(12,5),(12,1),(13,2),(13,3),(14,2),(14,4),(15,5);
+
+-- Câu 3. Viết lệnh sql để có thể lấy thông tin những bản ghi của projects và số lượng user của mỗi projects đó (count user)
+select  projects.project_id, projects.project_name, COUNT(project_users.project_id) as 'Số lượng User' from projects
+left join project_users on projects.project_id = project_users.project_id
+group by projects.project_id, projects.project_name, project_users.project_id;
+
+-- Câu 4. Viết lệnh sql để lấy ra danh sách các project của company có company_name = “monstar-lab” 
+select *from projects inner join companies
+on projects.company_id = companies.company_id
+where companies.company_name like 'monstar_lab';
+
+-- câu 5: Viết lệnh sql lấy ra danh sách các công ty có project có project_spend > 100
+select company_name from companies
+where company_id in (select company_id from projects where projects.projected_spend >100);
```

Comment:
sao không join mà lạị dùng where in thế này. Khi join bảng dc thì nên dùng join nhá, Về performance thì join sẽ tốt hơn